system/doc/programming_examples/list_comprehensions.xml

```diff
@@ -56,26 +56,43 @@
     <title>Quick Sort</title>
     <p>The well-known quick sort routine can be written as follows:</p>
     <code type="none"><![CDATA[
+sort([]) -> [];
+sort([_] = L) -> L;
 sort([Pivot|T]) ->
     sort([ X || X <- T, X < Pivot]) ++
     [Pivot] ++
-    sort([ X || X <- T, X >= Pivot]);
-sort([]) -> [].]]></code>
+    sort([ X || X <- T, X >= Pivot]).]]></code>
     <p>The expression <c><![CDATA[[X || X <- T, X < Pivot]]]></c> is the list of
       all elements in <c>T</c> that are less than <c>Pivot</c>.</p>
     <p><c><![CDATA[[X || X <- T, X >= Pivot]]]></c> is the list of all elements in
       <c>T</c> that are greater than or equal to <c>Pivot</c>.</p>
-      <p>A list sorted as follows:</p>
+    <p>With the algorithm above, a list is sorted as follows:</p>
       <list type="bulleted">
-       <item>The first element in the list is isolated
-       and the list is split into two sublists.</item>
-       <item>The first sublist contains
-      all elements that are smaller than the first element in
-      the list.</item>
-      <item>The second sublist contains all elements that are greater
-      than, or equal to, the first element in the list.</item>
-       <item>Then the sublists are sorted and the results are combined.</item>
-     </list>
+        <item>A list with zero or one element is trivially sorted.</item>
+        <item>For lists with more than one element:
+          <list type="ordered">
+            <item>The first element in the list is isolated as the pivot element.</item>
+            <item>The remaining list is partinioned into two sublists, such that:
```

Comment:
> Looks good except for a typo.

Ah... Funny that, since I started this in order to _fix_ a typo :sweat_smile:

> While you are it, perhaps it would be a good idea to mention that while this way to sort makes a neat example, a more efficient way to sort in Erlang is `lists:sort/1`?

Good idea, I'll see to it :+1:

Reply:
Ok, what do you think? (I'll squash the commits if you think it's ok.)